pyquil/tests/utils.py

```diff
@@ -1,10 +1,29 @@
+import asyncio
 import os
+import signal
+from contextlib import contextmanager
+from multiprocessing import Process
+
+import rpcq
+from qcs_api_client.client import QCSClientConfiguration
 
-from pyquil.api import Client
-from pyquil.quantum_processor import AbstractQuantumProcessor
-from pyquil.parser import parse
-from pyquil.api._abstract_compiler import AbstractCompiler
 from pyquil import Program
+from pyquil.api._abstract_compiler import AbstractCompiler
+from pyquil.parser import parse
+from pyquil.quantum_processor import AbstractQuantumProcessor
+
+
+@contextmanager
+def run_rpcq_server(server: rpcq.Server, endpoint: str):
+    def run_server():
+        server.run(endpoint=endpoint, loop=asyncio.new_event_loop())
+
+    proc = Process(target=run_server)
+    try:
+        proc.start()
```

Comment:
This is great. One thing that can help weird, system-dependent bugs from hanging tests - a connection check here prior to the yield but after a short sleep to make sure it's _actually_ listening.

Reply:
I'll give it a try :)

Reply:
Done.